src/mempool.cpp

```diff
@@ -359,13 +357,11 @@ bool MemPool::submit(const VbkBlock& blk, ValidationState& state) {
   }
 
   // stateful validation
-  if (tree_->vbk().getBlockIndex(blk.getHash())) {
+  if (!tree_->vbk().getBlockIndex(blk.getHash())) {
     // duplicate
-    return state.Invalid("pop-mempool-submit-vbkblock-stateful");
+    touchVbkBlock(blk, blk.getId());
   }
```

Comment:
I'd inverse it:
if (tree_->vbk().getBlockIndex(blk.getHash())) {
  return true;
}